server/app/services/question/types/QuestionDefinitionBuilder.java

```diff
@@ -19,41 +19,42 @@
 import services.question.types.TextQuestionDefinition.TextValidationPredicates;
 
 /**
- * Provides helper functions to build a QuestionDefinition.
+ * DEPRECATED. Provides helper functions to build a {@link QuestionDefinition}.
  *
- * <p>TODO(#4872): Remove this class in favor of QuestionDefinitionConfig.Builder.
+ * <p>TODO(#5271): Remove this class in favor of {@link QuestionDefinitionConfig.Builder}.
  */
 public final class QuestionDefinitionBuilder {
 
```

Comment:
As evident below, we stop storing the fields that are in `QuestionDefinitionConfig`, instead storing the `QuestionDefinitionConfig` itself to reduce duplication.